packages/components/src/internal/url/URLResolver.ts

```diff
@@ -49,15 +49,8 @@ export namespace URLService {
         const nextRoute = nextRouteState.location.pathname;
         const table = getRouteTable(store.getState());
 
-        if (table.has(nextRoute)) {
-            if (table.get(nextRoute) === true) {
-                next();
-            } else {
-                replace({
-                    pathname: table.get(nextRoute),
-                    query,
-                });
-            }
+        if (table.has(nextRoute) && table.get(nextRoute) !== true) {
```

Comment:
This is not the same logic as before. I would recommend:

```ts
if (table.has(nextRoute)) {
    if (table.get(nextRoute) !== true) {
        replace({ pathname: table.get(nextRoute), query });
    }
}
```

The key being we don't want to execute the other else below if the `table.has(nextRoute)` is ever true because that means the route has already been processed.

Reply:
Ah, right. I'll fix